pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -652,6 +658,45 @@ func (r *ReconcileLogStorage) getKibanaService(ctx context.Context) (*corev1.Ser
 	return &svc, nil
 }
 
+// "tigera-end-users" ConfigMap and tigera-elastic-users-credentials secret should be created only ones when external IdP is set and Elasticsearch uses basic license.
+// Data will be added to these objects by other repositories as end-user information is collected, these objects should not be overwritten.
+// If external IdP is not configured or if Elasticsearch is not using Basic license, delete these objects.
+func (r *ReconcileLogStorage) getElasticNativeUserConfig(ctx context.Context, dexCfg render.DexRelyingPartyConfig, licenseType render.ElasticLicenseType) render.ElasticNativeUsersConfig {
+	var esNativeUserConfig render.ElasticNativeUsersConfig
+	addUsersConfigMap := licenseType == render.ElasticLicenseTypeBasic && dexCfg != nil
+
+	objs := &[]runtime.Object{
+		&corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{Kind: "ConfigMap", APIVersion: "v1"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      render.OIDCUsersConfigMapName,
+				Namespace: render.ElasticsearchNamespace,
+			},
+		},
+		&corev1.Secret{
+			TypeMeta: metav1.TypeMeta{Kind: "Secret", APIVersion: "v1"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      render.OIDCUsersSecreteName,
+				Namespace: render.ElasticsearchNamespace,
+			},
+		},
+	}
+
+	cmErr := r.client.Get(ctx, types.NamespacedName{Name: render.OIDCUsersConfigMapName, Namespace: render.ElasticsearchNamespace}, &corev1.ConfigMap{})
+	secretErr := r.client.Get(ctx, types.NamespacedName{Name: render.OIDCUsersSecreteName, Namespace: render.ElasticsearchNamespace}, &corev1.Secret{})
+	if (cmErr != nil || secretErr != nil) && addUsersConfigMap {
```

Comment:
The render package allows you to return objects that need to be either created or deleted, this isn't the way to remove items, the second array in the declaration of `Objects() ([]runtime.Object, []runtime.Object)` is for object to delete.

The correct pattern for this (correct as in how the operator team decided to do this) would be look up the config map and secret and pass them to the render code. The render code can then decide whether or not to create the object or delete it. If these object don't exist then you would be passing nil to the render code, in which case the render code can take that as a sign to initialise those config maps.